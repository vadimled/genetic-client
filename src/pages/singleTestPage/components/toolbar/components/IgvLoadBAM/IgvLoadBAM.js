import React from 'react';
import { connect } from "react-redux";
import { Button, Icon } from 'antd';
import cn from 'classnames';
import style from './IgvLoadBAM.module.scss';
import {
  getIgvFetchBAMFileStatus,
  getBAMFileUrl
} from "Store/selectors";
import {
  fetchBAMFile
} from 'Actions/igvActions';

const IgvLoadBAM = (props) => {
  const {
    fetchBAMFileStatus,
    BAMFileUrl,
    fetchBAMFile
  } = props;

  return (
    <div className={cn(
      "toolbar-btn-wrapper",
      style["igv-btn"]
    )}>
      <Button
        type={fetchBAMFileStatus ? 'primary' : ''}
        onClick={fetchBAMFile.bind(null, BAMFileUrl)}
        disabled={fetchBAMFileStatus}
        className={fetchBAMFileStatus ? `progress progress--${fetchBAMFileStatus}` : ''}
        data-testid="open-igv-btn"
      >
        Load BAM{fetchBAMFileStatus === 3 && <Icon type="check" />}
      </Button>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    fetchBAMFileStatus: getIgvFetchBAMFileStatus(state),
    BAMFileUrl: getBAMFileUrl(state),
  };
};

function mapDispatchToProps(dispatch) {
  return {
    fetchBAMFile: data => dispatch(fetchBAMFile(data)),
  };
}

export default React.memo(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(IgvLoadBAM)
);
