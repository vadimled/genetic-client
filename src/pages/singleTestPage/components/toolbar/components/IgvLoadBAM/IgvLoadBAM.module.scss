.igv-btn :global {
  width: 114px;

  .ant-btn {
    &.progress {
      &::after {
        content: "";
        position: absolute;
        bottom: 0px;
        left: 0px;
        height: 4px;
        width: 100%;
        background-repeat: no-repeat;
      }

      &--2 {
        &::after {
          background-image: url("../../../../../../assets/btn-progress-part.svg");
          background-position: 0px calc(100% - 1px);
        }
      }

      &--3 {
        &::after {
          background-image: url("../../../../../../assets/btn-progress-full.svg");
          background-position: 0px calc(100% - 1px);
        }
      }

      & > span + .anticon {
        margin-left: 5px;
        font-size: 11px;
        svg {
          fill: white;
        }
      }
    }
  }
}
