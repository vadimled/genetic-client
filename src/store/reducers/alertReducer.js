import createReducer from "./createReducer";
import actionsTypes from "../actionsTypes";

const initialState = {
  status: null, // error | warning | default,
  title: '',
  message: ''
};

const tableReducer = createReducer(initialState, {
  [actionsTypes.SET_ALERT]: (state, { payload }) => {
    if (!payload) payload = {};
    const { status = null, title = '', message = ''} = payload;

    return {
      status,
      title,
      message
    };
  }
});

export default tableReducer;
