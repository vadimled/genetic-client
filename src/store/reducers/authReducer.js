import _ from "lodash";
import actionsTypes from '../actionsTypes';
import createReducer from "./createReducer";

const initialState = {
  isLoggedIn: false,
  status: null,
  serverError: null,
  userEmail: null
};

const authReducer = createReducer(initialState, {
  [actionsTypes.LOGIN_SUCCESS]: (state, {payload}) => {

    return {
      ...state,
      isLoggedIn: true,
      ...payload
    };
  },

  [actionsTypes.LOGIN_RESULT]: (state, {payload}) => {
    return {
      ...state,
      isLoggedIn: !_.has(payload, "message"),
      serverError: null,
      ...payload
    };
  },

});


export default authReducer;
