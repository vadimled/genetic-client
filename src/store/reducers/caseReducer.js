import createReducer from "./createReducer";
import actionsTypes from "../actionsTypes";

const initialState = {
  gsid: null,
  panel_type: null,
  test_id: null,
  showTumorInfo: false,
  mutation_types: [],
  tumorInfo: {
    type: "",
    location: "",
    percent: -1
  }
};

const caseReducer = createReducer(initialState, {
  [actionsTypes.SET_CASE_DATA]: (state, { payload }) => {
    return {
      ...state,
      ...payload
    };
  },

  [actionsTypes.SET_TUMOR_INFO_MODE]: (state, { payload }) => {
    return {
      ...state,
      showTumorInfo: payload
    };
  },

  [actionsTypes.SET_TUMOR_INFO]: (state, { payload }) => {
    // const {type: [val](|| location || percent)} = payload;
    return {
      ...state,
      tumorInfo: {
        ...state.tumorInfo,
        ...payload
      }
    };
  }
});

export default caseReducer;
