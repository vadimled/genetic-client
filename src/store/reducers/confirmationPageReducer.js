import createReducer from "./createReducer";
import actionsTypes from "../actionsTypes";

const initialState = {
  metaData: null
};

const confirmationPageReducer = createReducer(initialState, {
  [actionsTypes.SET_CONFIRMATION_PAGE_METADATA]: (state, { payload }) => {
    return {
      ...state,
      metaData: payload
    };
  }
});

export default confirmationPageReducer;
