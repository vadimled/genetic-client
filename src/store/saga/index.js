import {
  takeEvery,
  takeLatest
} from "redux-saga/effects";
import types from "Store/actionsTypes";

import {
  fetchTableData,
  fetchTestDataGenerator,
  fetchBAMFileGenerator,
  goToChrPositionIgvGenerator,
  sendForConfirmationGenerator,
  resultConfigLoadHgvsGenerator,
  resultConfigAddResultGenerator,
  resultConfigEditResultGenerator,
  fetchVariantDataGenerator,
  sendVariantClassGenerator,
  handleZygositySaga
} from "./sagas";

export function* watchSaga() {
  yield takeLatest(types.FETCH_TEST_DATA, fetchTestDataGenerator);
  yield takeLatest(types.FETCH_BAM_FILE, fetchBAMFileGenerator);
  yield takeLatest(types.GO_TO_CHR_POSITION_IGV, goToChrPositionIgvGenerator);
  yield takeLatest(types.SEND_FOR_CONFIRMATION, sendForConfirmationGenerator);
  yield takeEvery(types.FETCH_TABLE_DATA, fetchTableData);
  yield takeEvery(types.HANDLE_ZYGOSITY, handleZygositySaga);
  yield takeLatest(types.RESULT_CONFIG_LOAD_HGVS, resultConfigLoadHgvsGenerator);
  yield takeLatest(types.RESULT_CONFIG_ADD_RESULT, resultConfigAddResultGenerator);
  yield takeLatest(types.RESULT_CONFIG_EDIT_RESULT, resultConfigEditResultGenerator);
  yield takeLatest(types.FETCH_VARIANT_DATA, fetchVariantDataGenerator);
  yield takeLatest(types.SEND_VARIANT_CLASS, sendVariantClassGenerator);
}
