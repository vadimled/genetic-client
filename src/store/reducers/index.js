import { combineReducers } from "redux";
import filtersReducer from "./filtersReducer";
import tableReducer from "./tableReducer";
import variantsReducer from "./variantsReducer";
import igvReducer from "./igvReducer";
import confirmationReducer from "./confirmationReducer";
import alertReducer from "./alertReducer";
import resultConfigReducer from "./resultConfigReducer";
import variantPageReducer from "./variantPageReducer";
import testReducer from "./testReducer";
import testsPageReducer from "./testsPageReducer";
import evidenceConfigReducer from "./evidenceConfigReducer";
import confirmationPageReducer from "./confirmationPageReducer";
import coveragePageReducer from "./coveragePageReducer";
import finalReportReducer from "./finalReportReducer";

export default combineReducers({
  test: testReducer,
  tests: testsPageReducer,
  filters: filtersReducer,
  table: tableReducer,
  variants: variantsReducer,
  igv: igvReducer,
  confirmation: confirmationReducer,
  alert: alertReducer,
  resultConfig: resultConfigReducer,
  variantPage: combineReducers({
    pageData: variantPageReducer,
    evidenceConfig: evidenceConfigReducer
  }),
  confirmationPage: confirmationPageReducer,
  coveragePage: coveragePageReducer,
  finalReport: finalReportReducer
});

