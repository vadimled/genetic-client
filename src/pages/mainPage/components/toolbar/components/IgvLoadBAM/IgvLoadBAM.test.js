import React from "react";
import { createStore, applyMiddleware } from "redux";
import { fireEvent } from "@testing-library/react";
import "jest-dom/extend-expect";
import IgvLoadBAM from './IgvLoadBAM';
import reducers from "Store/reducers";
// import axiosMock from 'axios';

import createSagaMiddleware from 'redux-saga';
import { watchSaga } from "Store/saga";

import { renderWithRedux } from "Utils/test_helpers";

describe('IgvLoadBAM', () => {
  xit('IgvLoadBAM snapshot', () => {
    const { asFragment } = renderWithRedux(<IgvLoadBAM />);
    expect(asFragment()).toMatchSnapshot();
  });

  xit('open igv btn', () => {
    const { getByTestId } = renderWithRedux(<IgvLoadBAM />);
    const openIgvBtn = getByTestId('open-igv-btn');

    expect(openIgvBtn).toHaveTextContent("Load BAM");
  });

  xit('handle open-igv-btn true', () => {
    const sagaMiddleware = createSagaMiddleware();
    const { getByTestId, store } = renderWithRedux(
      <IgvLoadBAM />,
      createStore(reducers, applyMiddleware(sagaMiddleware))
    );
    sagaMiddleware.run(watchSaga);

    const openIgvBtn = getByTestId('open-igv-btn');

    axiosMock.get = () => true;

    fireEvent.click(openIgvBtn);

    const igvStore = store.getState().igv;
    expect(igvStore.isIgvAlertShow).toEqual(false);
    expect(igvStore.fetchBAMFileStatus).toEqual(null);
    expect(igvStore.igvLastQuery).toEqual(null);
  });

  xit('handle open-igv-btn false', () => {
    const sagaMiddleware = createSagaMiddleware();
    const { getByTestId, store } = renderWithRedux(
      <IgvLoadBAM />,
      createStore(reducers, applyMiddleware(sagaMiddleware))
    );
    sagaMiddleware.run(watchSaga);

    const openIgvBtn = getByTestId('open-igv-btn');

    axiosMock.get = () => {throw new Error();};

    fireEvent.click(openIgvBtn);

    const igvStore = store.getState().igv;
    expect(igvStore.isIgvAlertShow).toEqual(true);
    expect(igvStore.fetchBAMFileStatus).toEqual(null);
    expect(igvStore.igvLastQuery).toBeDefined();
  });
});
