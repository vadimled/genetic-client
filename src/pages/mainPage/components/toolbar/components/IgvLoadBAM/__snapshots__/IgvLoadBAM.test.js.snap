// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IgvLoadBAM IgvLoadBAM snapshot 1`] = `
<DocumentFragment>
  <div
    class="toolbar-btn-wrapper igv-btn"
  >
    <button
      class="ant-btn"
      data-testid="open-igv-btn"
      type="button"
    >
      <span>
        Load BAM
      </span>
    </button>
  </div>
</DocumentFragment>
`;
